molr-mole-core/src/test/java/io/molr/mole/core/support/MissionPredicatesTest.java

```diff
@@ -0,0 +1,36 @@
+package io.molr.mole.core.support;
+
+import io.molr.commons.domain.RunState;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class MissionPredicatesTest {
+
+    @Test
+    public void runState() {
+        MissionPredicates<RunState> runStateValidator = MissionPredicates.runStatePredicate(RunState.FINISHED);
+        assertNotNull(runStateValidator);
+        assertEquals(MissionPredicates.class, runStateValidator.getClass());
+    }
+
+    @Test
+    public void or() {
+        MissionPredicates<RunState> runStateValidator = MissionPredicates.runStatePredicate(RunState.FINISHED).or(RunState.RUNNING);
+        assertTrue(runStateValidator.test(RunState.RUNNING));
+        assertTrue(runStateValidator.test(RunState.FINISHED));
+        assertFalse(runStateValidator.test(RunState.PAUSED));
+    }
+
+    @Test
+    public void and() {
+        MissionPredicates<RunState> runStateValidator1 = MissionPredicates.runStatePredicate(RunState.FINISHED).and(RunState.RUNNING);
```

Comment:
How could such an expression ever make sense? A MissionState only can have one RunState, so a logical AND will never be useful... I think there is a bit of misconception in the MissionPredicates class... I will put some comments there ...